### ISSUE TITLE
[CRASH] Object erased from the project file list remains on footage viewer

### ISSUE THREAD

**Issue by signedbelt5** [olive-editor/olive#1249]
**Commit Hash**
b9359ce5

**Platform**
Pop OS 20.04 (_Architecture:_ x86-64 _Environment:_ KDE Plasma 5.18.5)

**Summary**
Erasing a project from the project file list while it's still in the footage viewer causes a crash.

**Steps to Reproduce**
1. Import a file.
2. Send it to the footage viewer.
3. Move the playhead.
4. Erase the file from the project's file list.
<details>
<summary> **Backtrace** </summary>

``` gdb backtrace
Thread 1 "olive-editor" received signal SIGSEGV, Segmentation fault.
olive::Item::parent (this=this@entry=0x0) at /home/signedb7/Documents/olive/app/project/item/item.cpp:128
128       return parent_;
```
Then:
``` gdb backtrace
#0  olive::Item::parent() const (this=this@entry=0x0) at /home/signedb7/Documents/olive/app/project/item/item.cpp:128
#1  0x00005555557cbf31 in olive::ProjectViewModel::IndexOfChild(olive::Item*) const (this=this@entry=0x5555583bec60, item=item@entry=0x0)
    at /home/signedb7/Documents/olive/app/project/projectviewmodel.cpp:437
#2  0x00005555557cc06d in olive::ProjectViewModel::parent(QModelIndex const&) const (this=0x5555583bec60, child=...)
    at /home/signedb7/Documents/olive/app/project/projectviewmodel.cpp:83
#3  0x00007ffff5019986 in QTreeView::isIndexHidden(QModelIndex const&) const () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#4  0x00007ffff501c101 in QTreeView::visualRect(QModelIndex const&) const () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#5  0x00007ffff4faaf02 in  () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#6  0x00007ffff4fb0bef in QAbstractItemView::edit(QModelIndex const&, QAbstractItemView::EditTrigger, QEvent*) ()
    at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#7  0x00007ffff4fa8902 in QAbstractItemView::edit(QModelIndex const&) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#8  0x00005555558598fc in olive::ProjectExplorer::RenameTimerSlot() (this=0x5555583bec00)
    at /home/signedb7/Documents/olive/app/widget/projectexplorer/projectexplorer.cpp:229
#9  0x00007ffff4352458 in QMetaObject::activate(QObject*, int, int, void**) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#10 0x00007ffff435f51e in QTimer::timeout(QTimer::QPrivateSignal) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#11 0x00007ffff4352cf5 in QObject::event(QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#12 0x00007ffff4d40a66 in QApplicationPrivate::notify_helper(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#13 0x00007ffff4d4a0f0 in QApplication::notify(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#14 0x00007ffff432693a in QCoreApplication::notifyInternal2(QObject*, QEvent*) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#15 0x00007ffff437d8b0 in QTimerInfoList::activateTimers() () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#16 0x00007ffff437e19c in  () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#17 0x00007ffff248afbd in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#18 0x00007ffff248b240 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#19 0x00007ffff248b2e3 in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#20 0x00007ffff437e565 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#21 0x00007ffff43254db in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#22 0x00007ffff432d246 in QCoreApplication::exec() () at /lib/x86_64-linux-gnu/libQt5Core.so.5
#23 0x00005555556ca905 in main(int, char**) (argc=<optimized out>, argv=0x7fffffffde58) at /home/signedb7/Documents/olive/app/main.cpp:161
```
</details>

**Additional Information / Output**
There was another crash detected if steps 3 & 4 are inverted, and another while trying to close Olive with a missing file remaining on the viewer.

**Comment by ThomasWilshaw** [comment]
Probably linked to #1232 but I will double check

**Comment by itsmattkc** [comment]
Fixed in d552e7184510848ca43e9717b338b299325fd7eb
